### PR TITLE
Adds support for passing hits_addend to ratelimit grpc client

### PR DESCRIPTION
**Pull request by PeterL328**
Commit Message: Adds support for passing hits_addend to ratelimit grpc client for weighted ratelimit request.
Additional Description: Note: This only modifies the grpc client. The ratelimit filter's behavior remains unchanged in this PR.
Risk Level: Low
Testing: Unit test
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features:
[Optional Runtime guard:] N/A. Added default argument to 0 so ratelimit filter's behavior will remain unchanged.
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: test/extensions/filters/common/ratelimit/ratelimit_impl_test.cc [hunk 1]
```diff
@@ -168,6 +168,36 @@ TEST_F(RateLimitGrpcClientTest, Cancel) {
   client_.cancel();
 }
 
+TEST_F(RateLimitGrpcClientTest, WeightedRequest) {
```
Comment (adisuissa) [inline test/extensions/filters/common/ratelimit/ratelimit_impl_test.cc:171]: Please add a comment describing what this test validates.
Also consider renaming the test's name as weighted-request may have a different meaning (maybe something like `RequestWithHitsAddend`)

**Review by adisuissa — COMMENTED** [review]
Thanks!
Minor comment, but otherwise LGTM.

**Comment by PeterL328** [comment]
@zuercher do you mind taking a look? Need one more approval
😁

**Comment by PeterL328** [comment]
Or @mattklein123 if you can have a look 😃

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**